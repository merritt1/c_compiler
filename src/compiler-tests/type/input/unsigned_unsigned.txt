unsigned char a;
unsigned short b;
unsigned int c;
unsigned long d;
int main(void) {
a + b;
b + a;
a + c;
c + a;
a + d;
d + a;
b + c;
c + b;
c + d;
d + c;
a + b + c + d;
d + c + b + a;
}
