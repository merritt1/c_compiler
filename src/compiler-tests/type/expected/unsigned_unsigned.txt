=============== PARSE TREE ===============
unsigned char (a);
unsigned short int (b);
unsigned int (c);
unsigned long int (d);
signed int (main(void)) {
((signed int) a+(signed int) b);
((signed int) b+(signed int) a);
((unsigned int) a+(unsigned int) c);
((unsigned int) c+(unsigned int) a);
((unsigned long int) a+d);
(d+(unsigned long int) a);
((unsigned int) b+(unsigned int) c);
((unsigned int) c+(unsigned int) b);
((unsigned long int) c+d);
(d+(unsigned long int) c);
((unsigned long int) ((unsigned int) ((signed int) a+(signed int) b)+(unsigned int) c)+d);
(((d+(unsigned long int) c)+(unsigned long int) b)+(unsigned long int) a);
}
================= SYMBOLS ================
Symbol Table ID: 0
  a -> unsigned char
  b -> unsigned short int
  c -> unsigned int
  d -> unsigned long int
  main -> function(signed int, [void])

